app/controllers/criteria_controller.rb

```diff
@@ -0,0 +1,17 @@
+class CriteriaController < ApplicationController
+
+  # This method handles the drag/drop criteria sorting
+  def update_positions
+    @assignment = Assignment.find(params[:assignment_id])
+    @criteria = @assignment.get_criteria
+    position = 0
+
+    params[:criterion].each do |id|
```

Comment:
I will look into this. Also, if it's ok with you,I would like to make these changes in a separate pull request, since two of my other pull requests were based out of this one.